File: imagepicker/src/main/java/com/esafirm/imagepicker/model/ImageFactory.java
package com.esafirm.imagepicker.model;

import android.net.Uri;

import com.esafirm.imagepicker.helper.ImagePickerUtils;

import java.util.ArrayList;
import java.util.List;

public class ImageFactory {
    public static List<Image> singleListFromPath(Uri uri) {
        List<Image> images = new ArrayList<>();
        images.add(new Image(0, uri.getLastPathSegment(), uri.getPath(), uri));
        return images;
    }
}

File: imagepicker/src/main/java/com/esafirm/imagepicker/model/ImageFactory.kt
package com.esafirm.imagepicker.model

import android.content.ContentUris
import android.net.Uri
import com.esafirm.imagepicker.helper.ImagePickerUtils

object ImageFactory {
    @JvmStatic
    fun singleImage(uri: Uri, path: String): List<Image> {
        return listOf(Image(
            id = ContentUris.parseId(uri),
            name = ImagePickerUtils.getNameFromFilePath(path),
            path = path,
            mimeType = ImagePickerUtils.getMimeType(path),
        ))
    }
}
